src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -147,10 +152,18 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol e
                 conditionPartToCheck.GetLocation(),
                 whenPartToCheck.GetLocation());
 
+            var properties = ImmutableDictionary<string, string>.Empty;
+            var whenPartIsNullable = semanticModel.GetTypeInfo(whenPartMatch).Type?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
```

Comment:
💭 You used `Type?.` here, which means one of the following:

1. We are running this analyzer in cases where the semantic model is incomplete on the type(s) involved, and thus `Type` is null.
2. We are not running the analyzer in the previous condition, and thus `?.` could be trivially replaced with `.`.

I believe we *want* the second condition, especially with the recent work to enable analyzers in cases where the semantic model may be less complete. Therefore:

:bulb: It would be good to make sure the analyzer only runs when `Type` is non-null here, and then change this line from `?.` to `.`.